frontend/src/contexts/HostContext/HostContext.spec.js

```diff
@@ -141,136 +141,342 @@ describe('Context', () => {
+        };
+
+        const { eventHandlers } = setupEmitterMocks();
+
+        render(
+          <HostProvider>
+            <TestComponent />
+          </HostProvider>,
+        );
+
+        expect(screen.queryAllByTestId('player-staging').length).toBe(0);
+
+        act(() => {
+          eventHandlers.message({
+            event: 'player-connected',
+            payload: { playerId: 'TEST', playerName: 'TESTER' },
+          });
+        });
+
+        expect(socketInstance.sendMessage).toHaveBeenCalledWith({
+          event: 'update',
+          recipients: ['TEST'],
+          payload: {
+            gameState: 'connected',
+            message: {
+              big: 'Attempting to join lobby',
+              small: 'Please wait',
+            },
           },
-        },
+        });
       });
-    });
 
-    it('tells the middleware the player joined mid round if the game state is not waiting-for-players', () => {
-      const TestComponent = () => {
-        const { dispatch } = useContext(HostContext);
+      it('tells the middleware the player joined mid round if the game state is not waiting-for-players', () => {
+        const TestComponent = () => {
+          const { dispatch } = useContext(HostContext);
 
-        useEffect(() => {
-          dispatch({ type: 'DEAL_WHITE_CARDS', payload: {} });
-        }, []);
+          useEffect(() => {
+            dispatch({ type: 'DEAL_WHITE_CARDS', payload: {} });
+          }, []);
 
-        return <div />;
-      };
+          return <div />;
+        };
 
-      const { eventHandlers } = setupEmitterMocks();
+        const { eventHandlers } = setupEmitterMocks();
 
-      render(
-        <HostProvider>
-          <TestComponent />
-        </HostProvider>,
-      );
+        render(
+          <HostProvider>
+            <TestComponent />
+          </HostProvider>,
+        );
 
-      act(() => {
-        eventHandlers.message({
-          event: 'player-connected',
-          payload: { playerId: 'TEST' },
+        act(() => {
+          eventHandlers.message({
+            event: 'player-connected',
+            payload: { playerId: 'TEST', playerName: 'TESTER' },
+          });
         });
-      });
 
-      expect(socketInstance.sendMessage).toHaveBeenCalledWith({
-        event: 'update',
-        recipients: ['TEST'],
-        payload: {
-          gameState: 'connected',
-          message: {
-            big: 'A round is in progress',
-            small: 'You will join the next round automatically',
+        expect(socketInstance.sendMessage).toHaveBeenCalledWith({
+          event: 'update',
+          recipients: ['TEST'],
+          payload: {
+            gameState: 'connected',
+            message: {
+              big: 'A round is in progress',
+              small: 'You will join the next round automatically',
+            },
+            removeLoading: 'joining-lobby',
           },
-          removeLoading: 'joining-lobby',
-        },
+        });
```

Comment:
Note to make reviewing easier: I didn't change anything here other than moving it into another describe block to group all of these (now many) `player-connected` related tests together.